packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/create-virtual-cluster.ts

```diff
@@ -0,0 +1,153 @@
+ */
+export class EksClusterInput {
+
+  /**
+   * Use an EKS Cluster for the EKS Cluster name
+   *
+   * @param cluster - An EKS cluster
+   * @returns The name of the EKS Cluster
+   */
+  static fromCluster(cluster: eks.ICluster): EksClusterInput {
+    return new EksClusterInput(cluster.clusterName);
+  }
+
+  /**
+   * Use a Task Input for the cluster name.
+   *
+   * @param taskInput Task Input object that accepts multiple types of payloads, in this case a literal string.
+   * @returns The value of a Task Input object, or a literal string.
+   */
+  static fromTaskInput(taskInput: sfn.TaskInput): EksClusterInput {
+    return new EksClusterInput(taskInput.value);
+  }
+
+  /**
+   * Initializes the clusterName
+   *
+   * @param clusterName The name of the EKS Cluster
+   */
+  private constructor(readonly clusterName: string) { }
+}
+
+/**
+ * Properties to define a EMR Containers CreateVirtualCluster Task on an EKS cluster
+ */
+export interface EmrContainersEksCreateVirtualClusterProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * EKS Cluster or task input that contains the name of the cluster
+   */
+  readonly eksCluster: EksClusterInput;
+
+  /**
+   * The namespace of an EKS cluster
+   *
+   * @default - 'default'
+   */
+  readonly eksNamespace?: string;
+
+  /**
+   * Name of the virtual cluster that will be created.
+   * @default - the id of the state machine execution that runs this task
+   */
+  readonly virtualClusterName?: string;
+
+  /**
+   * The tags assigned to the virtual cluster
+   *
+   * @default {}
+   */
+  readonly tags?: { [key: string]: string };
+}
+
+/**
+ * Task that creates an EMR Containers virtual cluster from an EKS cluster
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-emr-eks.html
+ */
+export class EmrContainersEksCreateVirtualCluster extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EmrContainersEksCreateVirtualClusterProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, EmrContainersEksCreateVirtualCluster.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('emr-containers', 'createVirtualCluster', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        Name: this.props.virtualClusterName ?? sfn.JsonPath.stringAt('$$.Execution.Id'),
```

Comment:
Where is `Execution.Id` coming from?

Reply:
from the execution [context object](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html)